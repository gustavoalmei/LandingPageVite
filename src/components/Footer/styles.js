import { styled, css } from 'styled-components';
import { Container as TextComponet } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../../components/SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.xsmall};
    }

    & > ${TextComponet} {
      font-size: ${theme.fonts.sizes.xsmall};
    }

    & > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
