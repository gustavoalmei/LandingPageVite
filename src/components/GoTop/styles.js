import { styled, css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;

    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.7;
    border-radius: 50%;
  `}
`;
