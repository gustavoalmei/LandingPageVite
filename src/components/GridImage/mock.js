export const mock = {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    [
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573996/photo_1730755803563_284a0bb30af2_056b06de14.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/eecc7d26262f6a1a80b325d916019656_85279db6cf.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/92eaac8133eb7bd2498bf4a7ab32fc13_b0c64273ab.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/8aed075b2259a6f2bace5c4924ceb0a3_cd7e3d5b77.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/92313bc1e9f73846662edc5681ad4408_f2159488ca.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/pexels_souvenirpixels_417074_9e0d95be44.jpg',
        altText: '',
      },
    ],
  ],
};
