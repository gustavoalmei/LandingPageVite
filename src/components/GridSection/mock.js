export const mock = {
  background: false,
  title: 'my grid',
  description:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
  grid: [
    {
      title: 'Teste 1',
      description:
        "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
    },
    {
      title: 'Teste 2',
      description:
        "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
    },
    {
      title: 'Teste 3',
      description:
        "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
    },
  ],
};
