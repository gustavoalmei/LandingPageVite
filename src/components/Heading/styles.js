import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.large};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1.withConfig({
  // withConfig permite configurar como o Styled Components manipula props específicas.
  shouldForwardProp: (prop) => !['colorDark', 'uppercase'].includes(prop),
  // shouldForwardProp é uma função que decide quais props serão passadas para o DOM. Nesse caso, estamos dizendo que colorDark não deve ser repassado.
})`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
  `}
`;
