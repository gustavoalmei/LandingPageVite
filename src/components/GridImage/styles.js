import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    height: 100%;
    object-fit: cover;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.3) rotate(10deg);
    }
  `}
`;
