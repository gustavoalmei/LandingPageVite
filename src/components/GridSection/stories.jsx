import GridSection from '.';
import { mock } from './mock';

export default {
  title: 'GridSection',
  component: GridSection,
  args: mock,
  argTypes: {
    children: { type: 'string' },
  },
};

export const Template = (args) => {
  return (
    <div>
      <GridSection {...args} />
    </div>
  );
};
