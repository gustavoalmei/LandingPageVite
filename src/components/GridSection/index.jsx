import P from 'prop-types';
import React from 'react';
import { Heading } from '../Heading';
import SectionBackground from '../SectionBackground';
import TextComponet from '../TextComponent';
import * as S from './styles';

export default function GridSection({ title, description, grid, background }) {
  return (
    <SectionBackground backgroundBool={background}>
      <S.Container>
        <Heading size="huge" uppercase colorDark={!background} as="h1">
          {title}
        </Heading>
        <TextComponet>{description}</TextComponet>
        <S.Grid>
          {grid.map((el) => (
            <S.GridElement key={el.title}>
              <Heading size="medium" colorDark={!background} as="h2">
                {el.title}
              </Heading>
              <TextComponet>{el.description}</TextComponet>
            </S.GridElement>
          ))}
        </S.Grid>
      </S.Container>
    </SectionBackground>
  );
}

GridSection.propTypes = {
  title: P.string.isRequired,
  description: P.string.isRequired,
  background: P.bool,
  grid: P.arrayOf(
    P.shape({
      title: P.string.isRequired,
      description: P.string.isRequired,
    }),
  ).isRequired,
};
