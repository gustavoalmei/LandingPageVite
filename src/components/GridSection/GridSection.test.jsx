import { RenderTheme } from '../../styles/handleTheme';
import GridSection from './index';
import { mock } from './mock';

describe('<GridSection/>', () => {
  test('Should be render', () => {
    const { container } = RenderTheme(<GridSection {...mock} />);

    expect(container).toMatchSnapshot();
  });
});
