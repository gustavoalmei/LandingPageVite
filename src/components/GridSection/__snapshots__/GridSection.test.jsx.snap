// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridSection/> Should be render 1`] = `
.c4 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c6 {
  font-size: 2.5rem;
}

.c2 .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
}

.c8 .c3 {
  position: relative;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  transform: rotate(10deg);
}

@media (max-width: 768px) {
  .c4 {
    font-size: 3.2rem;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3 c4"
        >
          my grid
        </h1>
        <div
          class="c5 c6"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
        </div>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h2
              class="c3 c9"
            >
              Teste 1
            </h2>
            <div
              class="c5 c6"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
            </div>
          </div>
          <div
            class="c8"
          >
            <h2
              class="c3 c9"
            >
              Teste 2
            </h2>
            <div
              class="c5 c6"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
            </div>
          </div>
          <div
            class="c8"
          >
            <h2
              class="c3 c9"
            >
              Teste 3
            </h2>
            <div
              class="c5 c6"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
